File: daily/2026/1260_Shift_2D_Grid.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        int m = grid.size();
        int n = grid[0].size();
        vector<vector<int>> ans(m, vector<int>(n));
        for (int i = 0; i < m * n; i++){
            int t = (i + k) % (m * n);
            ans[t / n][t % n] = grid[i / n][i % n];
        }
        return ans;
    }
};

int main(){
    vector<vector<int>> grid = {{1,2,3},{4,5,6},{7,8,9}};
    int k = 1;
    // vector<vector<int>> grid = {{3,8,1,9},{19,7,2,5},{4,6,11,10},{12,0,21,13}};
    // int k = 4;
    // vector<vector<int>> grid = {{1,2,3},{4,5,6},{7,8,9}};
    // int k = 9;
    // vector<vector<int>> grid = {{1},{2},{3},{4},{7},{6},{5}};
    // int k = 2;
    Solution S;
    vector<vector<int>> ans = S.shiftGrid(grid, k);
    for (auto i : ans){
        for (auto j : i){
            cout << j << " ";
        }
        cout << endl;
    }
    return 0;
}

File: daily_python/1260_Shift_2D_Grid.py
import os
import sys
from typing import List

class Solution:
    def gridSwap(self, grid: List[List[int]], start: int, end: int) -> None:
        m, n = len(grid), len(grid[0])
        while start < end:
            grid[start // n][start % n], grid[end // n][end % n] = grid[end // n][end % n], grid[start // n][start % n]
            start += 1
            end -= 1
        return
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        m, n = len(grid), len(grid[0])
        k %= m * n
        self.gridSwap(grid, 0, m * n - 1)
        self.gridSwap(grid, 0, k - 1)
        self.gridSwap(grid, k, m * n - 1)
        return grid
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            grid = [list(map(int, row.split(','))) for row in lines[0].split('=')[1].strip()[2:-3].split('},{')]
            k = int(lines[1].split('=')[1].strip()[:-1])
            dataset.append((grid, k))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for grid, k in dataset:
        results.append(solution.shiftGrid(grid, k))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')
